pkg/apis/core/v1beta1/types.go

```diff
@@ -362,6 +383,124 @@ type StorageProfileList struct {
 	Items []StorageProfile `json:"items"`
 }
 
+// DataSource references an import/clone source for a DataVolume
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type DataSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataSourceSpec   `json:"spec"`
+	Status DataSourceStatus `json:"status,omitempty"`
+}
+
+// DataSourceSpec defines specification for DataSource
+type DataSourceSpec struct {
+	// Source is the source of the data referenced by the DataSource
+	Source DataSourceSource `json:"source"`
+}
+
+// DataSourceSource represents the source for our DataSource
+type DataSourceSource struct {
+	PVC *DataSourceSourcePVC `json:"pvc"`
+}
+
+// DataSourceSourcePVC provides the parameters to refer an existing PVC
+type DataSourceSourcePVC = DataVolumeSourcePVC
```

Comment:
Ah, you never stop learning. Didn't know we can do that in Go: https://golang.org/ref/spec#Type_declarations
Reading that I don't think this additional alias is needed and we can use `DataVolumeSourcePVC` above instead of this kind of unergonomic name.

Reply:
Fixed